orthoinference/src/main/java/org/reactome/orthoinference/PathwaysInferrer.java

```diff
@@ -168,40 +166,31 @@ private static void addInferredEventsToInferredPathways() throws Exception {
 	private static void inferPrecedingEvents() throws Exception
 	{
 		Set<GKInstance> seenPrecedingEvent = new HashSet<>();
-		for (GKInstance inferrableEventInst : updatedInferrableHumanEvents)
-		{
-			if (!seenPrecedingEvent.contains(inferrableEventInst))
-			{
-				if (inferrableEventInst.getAttributeValue(precedingEvent)!= null)
-				{
+		for (GKInstance inferrableEventInst : updatedInferrableHumanEvents) {
+			if (!seenPrecedingEvent.contains(inferrableEventInst)) {
+				if (inferrableEventInst.getAttributeValue(precedingEvent) != null) {
```

Comment:
These two conditions could be combined with `&&` if `seenPrecedingEvent.add(inferrableEventInst);` (line 198) is moved outside the inner if condition.  Since `seenPrecedingEvent` is a set, it won't cause a problem if you add an instance that is already present in the set.

Reply:
You may also want to consider refactoring the body of the if statement to its own method so that you have the operation occurring for each `inferrableEventInst` isolated.